### DISCUSSION TITLE
how can I get the token of my Roborock S7?

### DISCUSSION THREAD

**Discussion by Gil80** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#451] [Q&A]
After reading the readme, I could only get tokens for my xiaomi devices, but not for the roborock vacuum.
is it even possible?

thanks!

**Comment by PiotrMachowski** [comment]
You can get token for Roborock vacuum if it is connected to Xiaomi Home app

**Reply by Gil80** [reply]
Do I need to remove it from the roborock app before adding it to the xiaomi mi home app?

**Reply by PiotrMachowski** [reply]
Yes, but you can't go back to Roborock app. You can use [this integration](https://github.com/humbertogontijo/homeassistant-roborock) without switching to Xiaomi Home
